### ISSUE TITLE
UWF_NULL_FIELD not reported when null initializer is wrapped in a redundant cast

### ISSUE THREAD

**Issue by Emilyaxe** [spotbugs/spotbugs#4034]
**Describe the bug**
`UWF_NULL_FIELD` is reported for a field initialized with `null`, but the warning disappears when it is written as `(String) null`, even though both are semantically identical. The cast should not affect the analysis, so this leads to a false negative.

**To Reproduce**

```java
// BEFORE — UWF_NULL_FIELD is reported on strVal
public class DefaultValueFields {
    private String strVal = null;

    public int compute(int input) {
        int result = 0;
        if (strVal != null) {
            result += strVal.length();
        }
        return result + input;
    }
}
```

```java
// AFTER — UWF_NULL_FIELD is NOT reported, although strVal is still only ever null
public class DefaultValueFields {
    private String strVal = (String) null;

    public int compute(int input) {
        int result = 0;
        if (strVal != null) {
            result += strVal.length();
        }
        return result + input;
    }
}
```

Run:

```
spotbugs -textui -effort:max DefaultValueFields.class
```

**Expected behavior**
`UWF_NULL_FIELD` should be reported on `strVal` in both versions, since the field is only ever written as `null` (a cast of `null` to its declared type is still `null`).

**Actual behavior**
`UWF_NULL_FIELD` fires on the BEFORE version but is suppressed on the AFTER version purely because the `null` initializer is wrapped in a `(String)` cast.

**Environment**

- Java version: JDK 21 (Eclipse Temurin)
- SpotBugs Version: 4.9.8
- How do you use SpotBugs: CLI

**Comment by omer-cengel** [comment]
Hi!

I'd love to help! I'll work on this.

**Comment by omer-cengel** [comment]
Hi!

I opened a PR https://github.com/spotbugs/spotbugs/pull/4061  that fixes this issue. When you have time, could you please take a look and review it? Thanks!
